### PR TITLE
feat: eigenlayer incredible squaring blueprint and test

### PR DESCRIPTION
**Pull request by Tjemmmic**
## Summary
Adds full test for the Eigen Incredible Squaring Blueprint, Replaces #304, rebased to Main,

### CODE REVIEW

File: blueprints/incredible-squaring/src/eigenlayer.rs [hunk 1]
```diff
@@ -227,15 +223,16 @@ impl GadgetRunner for EigenlayerGadgetRunner<parking_lot::RawRwLock> {
...
             eth_rpc_url: http_endpoint.to_string(),
             eth_ws_url: ws_endpoint.to_string(),
-            bls_private_key_store_path: "./keystore/bls".to_string(),
-            ecdsa_private_key_store_path: "./keystore/ecdsa".to_string(),
-            incredible_squaring_service_manager_addr: Default::default(),
-            avs_registry_coordinator_addr: Default::default(),
-            operator_state_retriever_addr: Default::default(),
+            bls_private_key_store_path: "./../eigensdk-rs/test-utils/keystore/bls".to_string(),
+            ecdsa_private_key_store_path: "./../eigensdk-rs/test-utils/keystore/ecdsa".to_string(),
+            incredible_squaring_service_manager_addr: "0DCd1Bf9A1b36cE34237eEaFef220932846BCD82"
```
Comment (drewstone) [inline blueprints/incredible-squaring/src/eigenlayer.rs:228]: Where did these address come from?

File: blueprints/incredible-squaring/src/eigenlayer.rs [hunk 2]
```diff
@@ -227,15 +223,16 @@ impl GadgetRunner for EigenlayerGadgetRunner<parking_lot::RawRwLock> {
...
+                .to_string(),
+            avs_registry_coordinator_addr: "0B306BF915C4d645ff596e518fAf3F9669b97016".to_string(),
```
Comment (drewstone) [inline blueprints/incredible-squaring/src/eigenlayer.rs:230]: We should pull out config into the top of the file as constants.

File: macros/blueprint-proc-macro/src/eigenlayer/event_listener.rs [hunk 3]
```diff
@@ -37,55 +36,53 @@ pub(crate) fn generate_eigenlayer_event_handler(
...
-                ContractInstance::new(
-                    self.instance.address().clone(),
-                    self.instance.provider().clone(),
-                    abi_location,
-                )
-            })
-        }
+                self.contract_instance.get_or_init(|| {
+                    let instance_string = stringify!(instance_name);
+                    let abi_path = format!("../contracts/out/{instance_string}.sol/{instance_string}.json");
```
Comment (drewstone) [inline macros/blueprint-proc-macro/src/eigenlayer/event_listener.rs:55]: Worth making sure that this will work on a fresh blueprint from the template/CLI.

**Review by drewstone — APPROVED** [review]
LG, some small nits and things to ensure.

**Comment by Tjemmmic** [comment]
This PR is almost ready. Just the following left:
- Fix: submitting job result on-chain
- Finalize Test for running in CI

**Comment by drewstone** [comment]
Status update @Tjemmmic ?

**Comment by Tjemmmic** [comment]
> Status update @Tjemmmic ?

It's close, I'm trying to get it wrapped up today. I am currently debugging BLS aggregation. Then I just need to clean up before merge

**Pull request merged**